modules/mysql/collect.go

```diff
@@ -68,17 +68,25 @@ func (m *MySQL) collect() (map[string]int64, error) {
 	// TODO: perhaps make a decisions based on privileges? (SHOW GRANTS FOR CURRENT_USER();)
 	if m.doSlaveStatus {
 		if err := m.collectSlaveStatus(mx); err != nil {
-			m.Errorf("error on collecting slave status: %v", err)
-			// TODO: shouldn't disable on any error
-			m.doSlaveStatus = false
+			if errors.Is(err, context.DeadlineExceeded) {
+				m.Warning("Deadline exceeded during slave status collection")
+			} else {
+				m.Errorf("error on collecting slave status: %v", err)
+				// TODO: shouldn't disable on any error
+				m.doSlaveStatus = false
```

Comment:
```suggestion
				m.doSlaveStatus = !errors.Is(err, context.DeadlineExceeded)
```

Probably just this + change Errorf to Warningf. What do you think?